state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1969,6 +1943,275 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+			) {
+			// This allows the channel to be boosted again:
+			DepositChannelLookup::<T, I>::mutate(deposit_address, |details| {
+				if let Some(details) = details {
+					details.boost_status = BoostStatus::NotBoosted;
+				}
+			});
+		}
+
+		Ok(())
+	}
+
+	fn assemble_broker_fees(
+		broker_fee: Beneficiary<T::AccountId>,
+		affiliate_fees: Affiliates<AffiliateShortId>,
+	) -> Beneficiaries<T::AccountId> {
+		let primary_broker = broker_fee.account.clone();
+
+		if T::AccountRoleRegistry::has_account_role(&primary_broker, AccountRole::Broker) {
+			[broker_fee]
+				.into_iter()
+				.chain(affiliate_fees.into_iter().filter_map(
+					|Beneficiary { account: short_affiliate_id, bps }| {
+						if let Some(affiliate_id) = T::AffiliateRegistry::get_account_id(
+							&primary_broker,
+							short_affiliate_id,
+						) {
+							Some(Beneficiary { account: affiliate_id, bps })
+						} else {
+							// In case the entry not found, we ignore the entry, but process the
+							// swap (to avoid having to refund it).
+							Self::deposit_event(Event::<T, I>::UnknownAffiliate {
+								broker_id: primary_broker.clone(),
+								short_affiliate_id,
+							});
+
+							None
+						}
+					},
+				))
+				.collect::<Vec<_>>()
+				.try_into()
+				.expect(
+					"must fit since affiliates are limited to 1 fewer element than beneficiaries",
+				)
+		} else {
+			Self::deposit_event(Event::<T, I>::UnknownBroker { broker_id: primary_broker });
+			Default::default()
+		}
+	}
+
+	fn process_prewitness_deposit_inner(
+		amount: TargetChainAmount<T, I>,
+		asset: TargetChainAsset<T, I>,
+		deposit_details: <T::TargetChain as Chain>::DepositDetails,
+		deposit_address: Option<TargetChainAccount<T, I>>,
+		source_address: Option<ForeignChainAddress>,
+		action: ChannelAction<T::AccountId>,
+		broker: &T::AccountId,
+		boost_fee: u16,
+		boost_status: BoostStatus<TargetChainAmount<T, I>>,
+		channel_id: Option<u64>,
+		block_height: TargetChainBlockNumber<T, I>,
+		origin: DepositOrigin<T, I>,
+	) -> Option<BoostStatus<TargetChainAmount<T, I>>> {
+		if amount < MinimumDeposit::<T, I>::get(asset) {
+			// We do not process/record pre-witnessed deposits for amounts smaller
+			// than MinimumDeposit to match how this is done on finalisation
+			return None;
+		}
+
+		if let Some(tx_id) = deposit_details.deposit_id() {
+			if TransactionsMarkedForRejection::<T, I>::mutate(broker, &tx_id, |opt| {
+				match opt.as_mut() {
+					// Transaction has been reported, mark it as pre-witnessed.
+					Some(status @ TransactionPrewitnessedStatus::Unseen) => {
+						*status = TransactionPrewitnessedStatus::Prewitnessed;
+						true
+					},
+					// Pre-witnessing twice is unlikely but possible. Either way we don't want
+					// to change the status and we don't want to allow boosting.
+					Some(TransactionPrewitnessedStatus::Prewitnessed) => true,
+					// Transaction has not been reported, mark it as boosted to prevent further
```

Comment:
This comment is a little confusing to me, we're not really marking anything here? we're just continuing in the function (which might then return a Boosted status)

Reply:
(wans't introduced here, but I removed the confusing/inaccurate part as discussed)